zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineLoader.kt

```diff
@@ -81,15 +103,37 @@ class ZiplineLoader(
       }
     }
 
-    val manifest = Json.decodeFromString<ZiplineManifest>(manifestByteString.utf8())
+    return Json.decodeFromString(manifestByteString.utf8())
+  }
+
+  suspend fun load(
+    zipline: Zipline,
+    manifest: ZiplineManifest,
+  ) {
+    loadImpl(manifest, getFromResourcesCacheThenNetworkHook(), loadIntoZiplineHook(zipline))
+  }
 
-    load(zipline, manifest)
+  private fun downloadToDirectoryHook(downloadDirectory: Path) = { ziplineFile: ZiplineFile, ziplineFileBytes: ByteString, id: String, sha256: ByteString ->
```

Comment:
Maybe make an `ExecutionMode` interface with two implementations, `Buildtime` and `Runtime`. Then all the different bits are features of the interface.